client/src/main/java/com/kritsit/casetracker/client/domain/ui/controller/AdministratorController.java

```diff
@@ -252,6 +260,49 @@ public void updateTable(){
         staffList = FXCollections.observableArrayList(administratorService.getInspectors());
     }
     
+    private void resetPassword(){
+        SelectionModel<Staff> selection = staffTable.getSelectionModel();
+        Alert alert;
+        if(selection.getSelectedItem()==null){
+            alert = new Alert(AlertType.WARNING);
+            alert.setTitle("Reseting password");
+            alert.setHeaderText("Information");
+            alert.setContentText("Select a user");
+            alert.showAndWait();   
+            return;
+        }
+        String username = selection.getSelectedItem().getUsername();
+        String randomPass = randomPassword();
+        int hashedRandomPass = randomPass.hashCode();
+        Map<String, Object> inputMap = new HashMap<String, Object>();
+        inputMap.put("username", username);
+        inputMap.put("hashedRandomPass", hashedRandomPass);
+        
+        int result = administratorService.resetPassword(inputMap);
+        switch(result){
+        case 200 :
+            alert = new Alert(AlertType.INFORMATION);
+            alert.setTitle("Password reset");
+            alert.setHeaderText("Password reset succesfully");
+            alert.setContentText("new password: "+randomPass);
+            alert.showAndWait();
+            break;
+        
+        case 500 :
+            alert = new Alert(AlertType.ERROR);
+            alert.setTitle("Password reset");
+            alert.setHeaderText("Error during password reset");
+            alert.setContentText("Error occured on the server side");
+            alert.showAndWait();
+            break;
+          }  
+    }
+    
+    private String randomPassword() {
```

Comment:
This should rather be in the Administrator class. I try to keep everything that is not directly related to UI code in the other classes